server/app/services/applicant/question/Question.java

```diff
@@ -29,13 +28,15 @@ public abstract class Question {
 
   public Question(ApplicantQuestion applicantQuestion) {
     this.applicantQuestion = Preconditions.checkNotNull(applicantQuestion);
-    if (!validQuestionTypes().contains(applicantQuestion.getType())) {
+
+    QuestionType supportedQuestionType = applicantQuestion.getType();
+    if (!getClass().equals(supportedQuestionType.getSupportedQuestion())) {
       throw new RuntimeException(
           String.format(
-              "Question is not a question of the following types: [%s]: %s (type: %s)",
-              Joiner.on(", ").join(validQuestionTypes().stream().toArray()),
-              applicantQuestion.getQuestionDefinition().getQuestionPathSegment(),
-              applicantQuestion.getQuestionDefinition().getQuestionType()));
+              "%s is not equal to question type %s's supported %s",
+              getClass(),
+              supportedQuestionType,
+              supportedQuestionType.getSupportedQuestion().toString()));
     }
   }
```

Comment:
The new error message looks like:

```
java.lang.RuntimeException: Class public final class services.applicant.question.TextQuestion is not equal to question type ADDRESS's supported class class services.applicant.question.AddressQuestion
```

Reply:
suggestion: this might be clearer if there format string was
```
The question type \"%s\" is not equal to question type supported by \"%s\", which is %s
```

Reply:
Made a slight modification to your suggestion and committed.